appscale/tools/agents/azure_agent.py

```diff
@@ -283,14 +283,16 @@ def describe_instances(self, parameters, pending=False):
       vm_list = compute_client.virtual_machine_scale_set_vms.list(resource_group,
                                                                   vmss.name)
       for vm in vm_list:
-        instance_ids.append(vm.name)
-      network_interface_list = network_client.network_interfaces.\
-        list_virtual_machine_scale_set_network_interfaces(resource_group,
-                                                          vmss.name)
-      for network_interface in network_interface_list:
-        for ip_config in network_interface.ip_configurations:
-          public_ips.append(ip_config.private_ip_address)
-          private_ips.append(ip_config.private_ip_address)
+        network_interface_list = network_client.network_interfaces. \
+          list_virtual_machine_scale_set_vm_network_interfaces(resource_group,
+                                                               vmss.name,
+                                                               vm.instance_id)
+        for network_interface in network_interface_list:
+          for ip_config in network_interface.ip_configurations:
```

Comment:
These for loops make it seem possible for the VM to be added to the list multiple times. I think it would be clearer to ensure that the appending only happens at most once for each VM.

For example, you could generate a list of IP addresses associated with a VM and choose the first one (or continue to the next VM if none exist).

This would prevent issues in case Azure ever decides to start automatically giving machines multiple network interfaces or something.